lang/csharp/src/apache/main/Schema/RecordSchema.cs

```diff
@@ -165,10 +284,19 @@ private static Field createField(JToken jfield, int pos, SchemaNames names, stri
         private static void addToFieldMap(Dictionary<string, Field> map, string name, Field field)
         {
             if (map.ContainsKey(name))
-                throw new SchemaParseException("field or alias " + name + " is a duplicate name");
+                throw new AvroException("field or alias " + name + " is a duplicate name");
```

Comment:
Don't change exception type.  Potentially breaking.  You would want to mark the old method obsolete and recreate with new exception type.